federation-js/src/service/printCoreSchema.ts

```diff
@@ -309,54 +330,56 @@ export function printWithReducedWhitespace(ast: ASTNode): string {
 }
 
 /**
- * Federation change: print fieldsets for @key, @requires, and @provides directives
+ * Core change: print fieldsets for @join__field's @key, @requires, and @provides args
  *
  * @param selections
  */
 function printFieldSet(selections: readonly SelectionNode[]): string {
-  return `{ ${selections.map(printWithReducedWhitespace).join(' ')} }`;
+  return `${selections.map(printWithReducedWhitespace).join(' ')}`;
 }
 
 /**
- * Federation change: print @resolve, @requires, and @provides directives
+ * Core change: print @join__field directives
  *
  * @param field
  * @param parentType
  */
-function printFederationFieldDirectives(
+function printJoinFieldDirectives(
   field: GraphQLField<any, any>,
   parentType: GraphQLObjectType | GraphQLInterfaceType,
 ): string {
-  if (!field.extensions?.federation) return '';
+  let printed = ' @join__field(graph: ';
+  // Fields on the owning service do not have any federation metadata applied
+  // TODO: maybe make this metadata available? Though I think this is intended and we may depend on that implicity.
+  if (!field.extensions?.federation) {
+    if (parentType.extensions?.federation?.serviceName) {
+      return printed + `${parentType.extensions?.federation.serviceName.toUpperCase()})`;
+    }
+    return '';
+  }
 
   const {
     serviceName,
     requires = [],
     provides = [],
   }: FederationField = field.extensions.federation;
 
-  let printed = '';
-  // If a `serviceName` exists, we only want to print a `@resolve` directive
-  // if the `serviceName` differs from the `parentType`'s `serviceName`
-  if (
-    serviceName &&
-    serviceName !== parentType.extensions?.federation?.serviceName
-  ) {
-    printed += ` @resolve(graph: "${serviceName}")`;
+  if (serviceName) {
+    printed += serviceName.toUpperCase();
   }
```

Comment:
The way this currently works we end up generating invalid SDL when the `serviceName` is `undefined` or `null` (which seems to happen for value types). I tried to fix this in a09e4339b6232ce400a40edc9e0694b681f4b7a2, but handling value types correctly may require more thought (see 081c59e75ff19df4433702bbeacd20a7d635e3f3).